### PR TITLE
Loads roles and users from config

### PR DESCRIPTION
**Pull request by 1maple1**

### CODE REVIEW

File: src/app/beer_garden/role.py [hunk 1]
```diff
@@ -54,14 +55,26 @@ def delete_role(role: Role = None, role_name: str = None, role_id: str = None) -
...
+    roles_yaml = "example_configs/roles.yaml"
+    with open(roles_yaml, "r") as config:
+        return yaml.safe_load(config)
+
 def rescan():
     """ Rescan the roles configuration file"""
-    pass
+    roles_config = load_roles_config()
+    for role in roles_config:
+        role = Role(name=role.get("name"), permission=role.get("permission"))
```
Comment (TheBurchLog) [inline src/app/beer_garden/role.py]: Missing the various scopes for roles. These are lists of string values.
https://github.com/beer-garden/brewtils/blob/User_Model/brewtils/models.py#L1666

File: src/app/beer_garden/role.py [hunk 2]
```diff
@@ -54,14 +55,32 @@ def delete_role(role: Role = None, role_name: str = None, role_id: str = None) -
...
+    roles_config = load_roles_config()
+    for role in roles_config:
+        role = Role(name=role.get("name"), permission=role.get("permission"),
+                    description=role.get("description"),
+                    scope_gardens=role.get("scope_gardens"),
+                    scope_namespaces=role.get("scope_namespaces"),
+                    scope_systems=role.get("scope_systems"),
+                    scope_instances=role.get("scope_instances"),
+                    scope_versions=role.get("scope_versions"),
+                    scope_commands=role.get("scope_commands"))
```
Comment (TheBurchLog) [inline src/app/beer_garden/role.py]: Should add a default value for all of these, so they are not required to be populated. The only required fields are `name` and `permission`. I think nulls will work, but if not, we can make scopes empty arrays and description an empty string.

Comment (1maple1) [reply src/app/beer_garden/role.py]: Get was used for the very reason that some of these will not be populated and will default to None much like in the brewtils model.

Comment (TheBurchLog) [reply src/app/beer_garden/role.py]: Yeah, I forgot you didn't need to provide defaults. This is all good then.

**Review by TheBurchLog — APPROVED** [review]

**Review by TheBurchLog — APPROVED** [review]

**Pull request merged**